### PR TITLE
[DPB][YANG] Fix cases when boolean is used in different literal cases

### LINKED ISSUE #9326: [DPB] DPB falls cause of invalid value in "has_global_scope" element
#### Description
The Dynamic Port Breakout feature is not working as expected on the latest master image

#### Steps to reproduce the issue:
1. `sudo config interface breakout Ethernet0 4x100G[50G] -f`

**NOTE:**
To reproduce this issue, a following PRs that fix other issues in SONiC should be merged into master:
1) Azure/sonic-buildimage/pull/9075
2) Azure/sonic-buildimage/pull/9276
#### Describe the results you received:
```
admin@sonic:~$ sudo config interface breakout Ethernet192 4x100G[50G] -f
Do you want to Breakout the port, continue? [y/N]: y

Running Breakout Mode : 1x400G[200G] 
Target Breakout Mode : 4x100G[50G]

Ports to be deleted : 
 {
    "Ethernet192": "400000"
}
Ports to be added : 
 {
    "Ethernet192": "100000",
    "Ethernet194": "100000",
    "Ethernet196": "100000",
    "Ethernet198": "100000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet192": "400000"
} 
Final list of ports to be added :  
 {
    "Ethernet192": "100000",
    "Ethernet194": "100000",
    "Ethernet196": "100000",
    "Ethernet198": "100000"
}

Note: Below table(s) have no YANG models:
KDUMP, SNMP, SNMP_COMMUNITY, 
libyang[0]: Invalid value "False" in "has_global_scope" element. (path: /sonic-feature:sonic-feature/FEATURE/FEATURE_LIST[name='bgp']/has_global_scope)
sonic_yang(3):Data Loading Failed:Invalid value "False" in "has_global_scope" element.
Data Loading Failed
Invalid value "False" in "has_global_scope" element.
ConfigMgmt Class creation failed
Failed to break out Port. Error: Failed to load the config. Error: ConfigMgmtDPB Class creation failed
```
#### Describe the results you expected:
config interface breakout should work without errors

#### Output of `show version`:
For testing, I manually built the sonic-image 15 November 2021 with last commits on the sonic-buildimage master and resolved issues which I mentioned in "NOTE" to "Steps to reproduce the issue".

### PR DESCRIPTION
**Pull request by mykolaxgerasymenko**
* Add boolean as typedef to sonic-types
* Fix boolean in sonic-feature yang model
* Fix boolean in sonic-flex_counter yang model

Signed-off-by: Mykola Gerasymenko <mykolax.gerasymenko@intel.com>

Fixes #9326

#### Why I did it
When we try execute DPB from CLI we have error:
`libyang[0]: Invalid value "False" in "has_global_scope" element. (path: /sonic-feature:sonic-feature/FEATURE/FEATURE_LIST[name='bgp']/has_global_scope)`
The reason for this issue is that has_global_scope and other have been stored in redis database with value False or True form capital letter:
```
   "FEATURE":{
      "bgp":{
         "auto_restart":"enabled",
         "has_global_scope":"False",
         "has_per_asic_scope":"True",
         "has_timer":"False",
         "high_mem_alert":"disabled",
         "state":"enabled"
      }
```
But yang model support boolean just in lowercase letters (https://datatracker.ietf.org/doc/html/rfc6020#section-9.5.1).
#### How I did it
Added boolean to sonic-types as typedef with different literal cases.

#### How to verify it
Run the command config interface breakout <breakout_mode>

**NOTE:**
To verify this fix, the following PRs that fix other problems in SONiC must be merged into master:
1) Azure/sonic-buildimage/pull/9075
2) Azure/sonic-buildimage/pull/9276
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

**Review by akokhan — DISMISSED** [review]

File: src/sonic-yang-models/yang-models/sonic-feature.yang [hunk 1]
```diff
@@ -42,22 +42,22 @@ module sonic-feature{
                 leaf has_timer {
                     description "This configuration identicates if there is
                                 timer associated to this feature";
-                    type boolean;
-                    default false;
+                    type stypes:boolean_type;
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-feature.yang:45]: >boolean_type [](http://example.com/codeflow?start=32&length=12)

Add some test cases for the new type?

Comment (mykolaxgerasymenko) [reply src/sonic-yang-models/yang-models/sonic-feature.yang:45]: Test cases for new boolean type were added. Please review.

**Comment by mykolaxgerasymenko** [comment]
@qiluo-msft Please review.

**Review by qiluo-msft — DISMISSED** [review]

**Comment by tw0byt3** [comment]
@mykolaxgerasymenko After the fix of this is done, the breakout fails due to the CRM yang model. I tried to accommodate the changes required. Post that, this is the error I am facing. 

root@localhost:~# config interface breakout Ethernet0 4x10G[25G] 
Do you want to Breakout the port, continue? [y/N]: y

Running Breakout Mode : 1x100G[50G,40G,25G,10G] 
Target Breakout Mode : 4x10G[25G]

Ports to be deleted : 
 {
    "Ethernet0": "100000"
}
Ports to be added : 
 {
    "Ethernet0": "10000",
    "Ethernet1": "10000",
    "Ethernet2": "10000",
    "Ethernet3": "10000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet0": "100000"
} 
Final list of ports to be added :  
 {
    "Ethernet0": "10000",
    "Ethernet1": "10000",
    "Ethernet2": "10000",
    "Ethernet3": "10000"
}
Note: Below table(s) have no YANG models:
KDUMP, SNMP, SNMP_COMMUNITY, XCVRD_LOG, 
Below Config can not be verified, It may cause harm to the system
 {}
Do you wish to Continue? [y/N]: y
Create Config to load in DB, Failed
'str' object has no attribute 'append'
Config Diff Generation failed
'str' object has no attribute 'append'
'str' object has no attribute 'append'
Port Deletion Failed
[ERROR] Port breakout Failed!!! Opting Out
Failed to break out Port. Error: 

Can someone help me identity there is this str object?

**Comment by mykolaxgerasymenko** [comment]
> @mykolaxgerasymenko After the fix of this is done, the breakout fails due to the CRM yang model. I tried to accommodate the changes required. Post that, this is the error I am facing.
> 
> root@localhost:~# config interface breakout Ethernet0 4x10G[25G] Do you want to Breakout the port, continue? [y/N]: y
> 
> Running Breakout Mode : 1x100G[50G,40G,25G,10G] Target Breakout Mode : 4x10G[25G]
> 
> Ports to be deleted : { "Ethernet0": "100000" } Ports to be added : { "Ethernet0": "10000", "Ethernet1": "10000", "Ethernet2": "10000", "Ethernet3": "10000" }
> 
> After running Logic to limit the impact
> 
> Final list of ports to be deleted : { "Ethernet0": "100000" } Final list of ports to be added : { "Ethernet0": "10000", "Ethernet1": "10000", "Ethernet2": "10000", "Ethernet3": "10000" } Note: Below table(s) have no YANG models: KDUMP, SNMP, SNMP_COMMUNITY, XCVRD_LOG, Below Config can not be verified, It may cause harm to the system {} Do you wish to Continue? [y/N]: y Create Config to load in DB, Failed 'str' object has no attribute 'append' Config Diff Generation failed 'str' object has no attribute 'append' 'str' object has no attribute 'append' Port Deletion Failed [ERROR] Port breakout Failed!!! Opting Out Failed to break out Port. Error:
> 
> Can someone help me identity there is this str object?

@tw0byt3 
Did you fix the boolean type for src/sonic-yang-models/yang-models/sonic-flex_counter.yang file as I fixed in this PR?

**Comment by tw0byt3** [comment]
@mykolaxgerasymenko Yes, I fixed the boolean issue similar to what you have fixed in this PR. The CRM yang model is missing srv6_my_sid_entry_high_threshold, srv6_my_sid_entry_low_threshold, srv6_my_sid_entry_threshold_type, srv6_nexthop_high_threshold, srv6_nexthop_low_threshold, srv6_nexthop_threshold_type.

**Review by akokhan — DISMISSED** [review]

File: src/sonic-yang-models/tests/files/sample_config_db.json [hunk 2]
```diff
@@ -1201,9 +1201,9 @@
         "FEATURE": {
             "bgp": {
                 "auto_restart": "enabled",
-                "has_global_scope": "false",
-                "has_per_asic_scope": "true",
-                "has_timer": "false",
+                "has_global_scope": "False",
+                "has_per_asic_scope": "True",
+                "has_timer": "False",
```
Comment (mykolaxgerasymenko) [inline src/sonic-yang-models/tests/files/sample_config_db.json:1206]: I have added these changes with "False" and "True" with a capital letter to make it clear that this option can also be used and it is valid.

**Comment by tw0byt3** [comment]
@mykolaxgerasymenko Any idea where can I find the 'str' object causing the above issue mentioned.

**Comment by mykolaxgerasymenko** [comment]
@tw0byt3 I faced a similar issue log when was problem with comparing boolean type in sonic-flex_counter.yang. But I fixed it in this PR. After this fix the DPB feature worked correctly.
I haven't tried using DPB after that, so it's possible that some new commits are causing the problem you mentioned.
Hopefully I'll try to reproduce this issue, but I have more priority task for now.
And it will be better if this PR will merge. In this case, the issues resolved in this PR will be excluded.

**Comment by mykolaxgerasymenko** [comment]
@qiluo-msft please merge

**Comment by qiluo-msft** [comment]
Could you solve the conflict?

**Comment by mykolaxgerasymenko** [comment]
> Could you solve the conflict?

Conflict has been resolved.

**Review by qiluo-msft — DISMISSED** [review]

**Comment by mykolaxgerasymenko** [comment]
@qiluo-msft please restart build CI. I do not have permissions.

**Comment by qiluo-msft** [comment]
I triggered the build. As a PR owner, you can try `/azpw run Azure.sonic-buildimage` on this page.

**Review by qiluo-msft — APPROVED** [review]

**Comment by mykolaxgerasymenko** [comment]
> I triggered the build. As a PR owner, you can try `/azpw run Azure.sonic-buildimage` on this page.

Thank you.

**Comment by judyjoseph** [comment]
Please raise a new PR for 202111 branch, as this patch cannot be cleanly cherry-picked.

**Pull request merged**